tableau_utilities/scripts/datasource.py

```diff
@@ -16,25 +16,33 @@ def datasource(args, server=None):
         args: An argparse args object
         server (TableauServer): A TableauServer object; optional
     """
+
+    # Set variables from args
     debugging_logs = args.debugging_logs
     datasource_path = args.file_path
+    datasource_id = args.id
+    datasource_name = args.name
+    project_name = args.project_name
+    include_extract = args.include_extract
+    save_tds = args.save_tds
+    location = args.location
+    persona = args.persona
+    delete = args.delete
+    column_name = args.column_name
     list_objects = args.list.title() if args.list else None
     color = Color()
     symbol = Symbol()
 
     # Downloads the datasource from Tableau Server if the datasource is not local
-    if args.location == 'online':
-        # Query Tableau Server for the datasource, use name/project if id was not provided
-        if not (args.id or args.name is not None and args.project_name is not None):
-            raise Exception('Cannot find datasource. Make sure --id or --name and --project_name are provided.')
```

Comment:
This exception can be covered in the CLI with this change https://github.com/hoverinc/tableau-utilities/pull/30/files#diff-57896026b09c4e5b973f80ac3fcb62efb131394da8119bd43b51cda5feabe49aR344

Reply:
Good catch!